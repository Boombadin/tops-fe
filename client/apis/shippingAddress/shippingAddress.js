import axios from 'axios'
import { baseUrl } from '../config'

const setShippingLocation = (regionId, districtId, subdistrictId, zipcode, customerId) => {
  const options = {
    method: 'POST',
    url: `${baseUrl}/shipping/current`,
    data: {
      regionId: regionId,
      districtId: districtId,
      subdistrictId: subdistrictId,
      zipcode: zipcode,
      customerId: customerId
    }
  }
  
  return axios(options)
    .then(response => response.data)
    .catch(e => {
      throw e
    })
}

export default {
  setShippingLocation
}