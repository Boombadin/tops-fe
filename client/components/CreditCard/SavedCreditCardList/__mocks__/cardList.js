export default [
  {
    bankIcon: '/images/payment/product-img-empty.svg',
    bankId: null,
    bankImage: '/images/payment/product-img-empty.svg',
    bankName: null,
    expiryDate: '12/33',
    expiryMonth: 12,
    expiryYear: 2033,
    id: '590925219668099073',
    isDefault: true,
    isExpired: false,
    maskedNumber: '411111XXXXXX1111',
    maskedNumberByStar: '**** **** **** 1111',
    type: 'Visa',
    typeIcon: '/icons/types-credit-card/visa-no-border.svg',
  },
  {
    bankIcon: '/images/payment/product-img-empty.svg',
    bankId: '10',
    bankImage: '/images/payment/product-img-empty.svg',
    bankName: 'KCC',
    expiryDate: '12/33',
    expiryMonth: 12,
    expiryYear: 2033,
    id: '604578118707118081',
    isDefault: false,
    isExpired: false,
    maskedNumber: '512345XXXXXX2346',
    maskedNumberByStar: '**** **** **** 2346',
    type: 'MasterCard',
    typeIcon: '/icons/types-credit-card/master-no-border.svg',
  },
];
