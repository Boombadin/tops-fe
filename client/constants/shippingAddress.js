export const requiredFields = [
  'firstname',
  'lastname',
  'telephone',
  'house_no',
  'region_id',
  'district_id',
  'subdistrict_id',
  'postcode'
];
