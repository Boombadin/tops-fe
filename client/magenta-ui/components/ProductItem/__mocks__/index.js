export * from '@client/magenta-ui/components/ProductItem/__mocks__/product';
