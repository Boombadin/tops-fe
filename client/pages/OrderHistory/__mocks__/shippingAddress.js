export const shippingAddress = [
  {
    shipping: {
      address: {
        address_type: 'shipping',
        city: 'กรุงเทพมหานคร',
        company: 'N/A',
        country_id: 'TH',
        email: 'soulcietycomsci+2@gmail.com',
        entity_id: 1243575,
        extension_attributes: {
          custom_attributes: [
            { attribute_code: 'moo', value: '1' },
            { attribute_code: 'soi', value: '1' },
            { attribute_code: 'address_name', value: 'name' },
            { attribute_code: 'house_no', value: '111' },
            { attribute_code: 'village_name', value: 'village_name' },
            { attribute_code: 'road', value: '12' },
            { attribute_code: 'district', value: 'บางรัก' },
            { attribute_code: 'subdistrict', value: 'บางรัก' },
          ],
        },
        firstname: 'test',
        lastname: 'test1',
        parent_id: 622300,
        postcode: '10500',
        region: 'กรุงเทพมหานคร',
        region_code: 'bkk',
        region_id: 534,
        street: ['N/A'],
        telephone: '0876612321',
      },
      method: 'tops_express',
    },
  },
];
