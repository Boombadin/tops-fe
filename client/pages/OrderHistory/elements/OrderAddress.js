import React from 'react';

import withLocales from '@client/hoc/withLocales';

export const OrderAddress = ({ translate, shipping_assignments }) => {
  const shippingAddress = shipping_assignments[0]?.shipping?.address || {};
  const customAttributes =
    shippingAddress?.extension_attributes?.custom_attributes || [];
  const shippingMethod = shipping_assignments[0]?.shipping?.method || '';

  let addressName = '';
  let address = '';

  customAttributes.map(resp => {
    if (
      resp?.attribute_code === 'moo' &&
      resp?.value !== '' &&
      resp?.value !== null
    ) {
      address += `${translate('shipping_address.prefix_moo')} ${resp.value} `;
    }
    if (
      resp?.attribute_code === 'soi' &&
      resp?.value !== '' &&
      resp?.value !== null
    ) {
      address += `${translate('shipping_address.prefix_soi')} ${resp?.value} `;
    }
    if (
      resp?.attribute_code === 'address_name' &&
      resp?.value !== '' &&
      resp?.value !== null
    ) {
      addressName = resp?.value;
    }
    if (
      (resp?.attribute_code === 'house_no' &&
        resp?.value !== '' &&
        resp?.value !== null) ||
      (resp?.attribute_code === 'village_name' &&
        resp?.value !== '' &&
        resp?.value !== null) ||
      (resp?.attribute_code === 'road' &&
        resp?.value !== '' &&
        resp?.value !== null) ||
      (resp?.attribute_code === 'district' &&
        resp?.value !== '' &&
        resp?.value !== null) ||
      (resp?.attribute_code === 'subdistrict' &&
        resp?.value !== '' &&
        resp?.value !== null)
    ) {
      address += `${resp?.value} `;
    }
  });

  address += `${shippingAddress?.region || ''} ${shippingAddress?.postcode ||
    ''}`;

  return (
    <span className="address">
      {shippingMethod === 'pickupatstore_tops' ? addressName : address}
    </span>
  );
};

export default withLocales(OrderAddress);
