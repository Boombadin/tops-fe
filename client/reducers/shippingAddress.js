import { find, get as prop } from 'lodash'
import ShippingAddressApi from '../apis/shippingAddress'
import RegionApi from '../apis/region'
import { getDefaultShippingAddressSelector } from '../selectors'

export const TYPES = {
  SET_CURRENT_PROVINCE: 'SET_CURRENT_PROVINCE',
  SET_CURRENT_DISTRICT: 'SET_CURRENT_DISTRICT',
  SET_CURRENT_SUB_DISTRICT: 'SET_CURRENT_SUB_DISTRICT',
  SET_CURRENT_ZIPCODE: 'SET_CURRENT_ZIPCODE'
}

export const saveShippingLocation = (regionId, districtId, subdistrictId, zipcode) => {
  return async (dispatch, getState) => {
    const { customer } = getState()
    const response = await ShippingAddressApi.setShippingLocation(regionId, districtId, subdistrictId, zipcode, prop(customer, 'items.id', 0))
    return response
  }
}

export const setCurrentShippingLocation = (regionId, districtId, subdistrictId, zipcode) => {
  return (dispatch) => {
    dispatch(setCurrentProvince(regionId))
    dispatch(setCurrentDistrict(districtId))
    dispatch(setCurrentSubDistrict(subdistrictId))
    dispatch(setZipcode(zipcode))
  }
}

export const setCurrentProvince = regionId => ({
  type: TYPES.SET_CURRENT_PROVINCE,
  payload: { regionId }
})

export const setCurrentDistrict = districtId => ({
  type: TYPES.SET_CURRENT_DISTRICT,
  payload: { districtId }
})

export const setCurrentSubDistrict = subdistrictId => ({
  type: TYPES.SET_CURRENT_SUB_DISTRICT,
  payload: { subdistrictId }
})

export const setZipcode = zipcode => ({
  type: TYPES.SET_CURRENT_ZIPCODE,
  payload: { zipcode }
})

const initialState = {
  province: '',
  district: '',
  subdistrict: '',
  zipcode: ''
}

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case TYPES.SET_CURRENT_PROVINCE: {
      const { regionId } = action.payload
      return { ...state, ...{ province: regionId } } 
    }

    case TYPES.SET_CURRENT_DISTRICT: {
      const { districtId } = action.payload
      return { ...state, ...{ district: districtId } } 
    }

    case TYPES.SET_CURRENT_SUB_DISTRICT: {
      const { subdistrictId } = action.payload
      return { ...state, ...{ subdistrict: subdistrictId } } 
    }

    case TYPES.SET_CURRENT_ZIPCODE: {
      const { zipcode } = action.payload
      return { ...state, ...{ zipcode: zipcode } } 
    }

    default:
      return state
  }
}

export default reducer
