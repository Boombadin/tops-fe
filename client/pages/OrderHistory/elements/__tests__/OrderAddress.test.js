import { render } from '@testing-library/react';
import React from 'react';

import { shippingAddress } from '../../__mocks__/shippingAddress';
import OrderAddress from '../OrderAddress';

jest.mock('@client/hoc/withLocales', () => {
  return Component => props => (
    <Component translate={jest.fn(key => key)} {...props} />
  );
});

describe('features/orderDetail/components/OrderAddress', () => {
  const props = {
    shipping_assignments: shippingAddress,
  };

  test('it should be match snapshot', () => {
    const { asFragment } = render(<OrderAddress {...props} />);
    expect(asFragment(<OrderAddress {...props} />)).toMatchSnapshot();
  });

  test('it should be match snapshot when region and postcode empty', () => {
    const customProps = {
      ...props,
      shipping_assignments: [
        {
          shipping: {
            ...shippingAddress[0].shipping,
            address: {
              ...shippingAddress[0].shipping.address,
              postcode: null,
              region: null,
            },
          },
        },
      ],
    };

    const { asFragment } = render(<OrderAddress {...customProps} />);
    expect(asFragment(<OrderAddress {...customProps} />)).toMatchSnapshot();
  });

  test('it should be match snapshot when address empty', () => {
    const customProps = {
      ...props,
      shipping_assignments: [
        {
          shipping: {
            ...shippingAddress[0].shipping,
            address: {},
          },
        },
      ],
    };

    const { asFragment } = render(<OrderAddress {...customProps} />);
    expect(asFragment(<OrderAddress {...customProps} />)).toMatchSnapshot();
  });

  test('it should be match snapshot when shipping empty', () => {
    const customProps = {
      ...props,
      shipping_assignments: [
        {
          shipping: {},
        },
      ],
    };

    const { asFragment } = render(<OrderAddress {...customProps} />);
    expect(asFragment(<OrderAddress {...customProps} />)).toMatchSnapshot();
  });

  test('it should be match snapshot when shipping method pickupatstore_tops', () => {
    const customProps = {
      shipping_assignments: [
        {
          shipping: {
            ...shippingAddress[0].shipping,
            method: 'pickupatstore_tops',
          },
        },
      ],
    };

    const { asFragment } = render(<OrderAddress {...customProps} />);
    expect(asFragment(<OrderAddress {...customProps} />)).toMatchSnapshot();
  });
});
