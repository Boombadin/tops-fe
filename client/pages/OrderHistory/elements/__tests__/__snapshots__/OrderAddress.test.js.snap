// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/orderDetail/components/OrderAddress it should be match snapshot 1`] = `
<DocumentFragment>
  <span
    class="address"
  >
    shipping_address.prefix_moo 1 shipping_address.prefix_soi 1 111 village_name 12 บางรัก บางรัก กรุงเทพมหานคร 10500
  </span>
</DocumentFragment>
`;

exports[`features/orderDetail/components/OrderAddress it should be match snapshot when address empty 1`] = `
<DocumentFragment>
  <span
    class="address"
  >
     
  </span>
</DocumentFragment>
`;

exports[`features/orderDetail/components/OrderAddress it should be match snapshot when region and postcode empty 1`] = `
<DocumentFragment>
  <span
    class="address"
  >
    shipping_address.prefix_moo 1 shipping_address.prefix_soi 1 111 village_name 12 บางรัก บางรัก  
  </span>
</DocumentFragment>
`;

exports[`features/orderDetail/components/OrderAddress it should be match snapshot when shipping empty 1`] = `
<DocumentFragment>
  <span
    class="address"
  >
     
  </span>
</DocumentFragment>
`;

exports[`features/orderDetail/components/OrderAddress it should be match snapshot when shipping method pickupatstore_tops 1`] = `
<DocumentFragment>
  <span
    class="address"
  >
    name
  </span>
</DocumentFragment>
`;
